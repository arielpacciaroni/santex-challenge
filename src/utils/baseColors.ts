export const baseColors = {
  "light-gray": "#f5f5f5",
  "highlight": "#eee",
  "success": "#15803d",
  "secondary": '#1c1917',
};