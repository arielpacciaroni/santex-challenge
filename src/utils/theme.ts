const colors = {
  'light-gray': '#f1f1f1',
  highlight: '#eee',
  blend: '#ccc',
  success: '#15803d',
  secondary: '#171717',
  information: '#bfdbfe',
  button: {
    'bg-primary': '#4d7c0f',
    'bg-primary-hover': '#65a30d',
    'text-primary': '#dcfce7',
    'bg-disabled': '#eee',
    'text-disabled': '#aaa',
    'bg-error': '#9a3412',
    'bg-error-hover': '#ea580c',
    'text-error': '#fff',
  },
};

const spacing = {
  small: '0.5rem',
  medium: '1rem',
};

export interface Theme {
  spacing: typeof spacing;
  colors: typeof colors;
}

export const theme = {
  spacing,
  colors,
};
