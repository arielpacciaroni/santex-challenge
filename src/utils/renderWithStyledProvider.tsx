import { render } from "@testing-library/react";
import { ReactNode } from "react"
import { ThemeProvider } from "styled-components";
import { theme } from "./theme";

export function renderWithStyledProvider(element: ReactNode) {
  return render(
    <ThemeProvider theme={theme}>
      {element}
    </ThemeProvider>
  )
}