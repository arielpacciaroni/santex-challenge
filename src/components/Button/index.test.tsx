import { Button } from '.';
import { renderWithStyledProvider } from '../../utils/renderWithStyledProvider';
import { fireEvent, screen } from '@testing-library/react';

describe('Button component', () => {
  it('should render a button', () => {
    const view = renderWithStyledProvider(
      <Button variant="primary">Hello world!</Button>
    );

    expect(screen.getByText('Hello world!')).toBeInTheDocument();
    expect(view).toMatchSnapshot();
  });

  it('should pass on the onClick button', async () => {
    const onClick = jest.fn();

    renderWithStyledProvider(
      <Button variant="primary" onClick={onClick}>
        Hello world!
      </Button>
    );

    fireEvent.click(screen.getByText('Hello world!'));

    expect(onClick).toHaveBeenCalled();
  });
});
