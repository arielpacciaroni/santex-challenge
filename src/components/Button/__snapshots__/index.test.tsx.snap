// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render a button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  border: none;
  outline: none;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  font-size: 0.875rem;
  font-weight: 600;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transition-delay: 100ms;
  transition-delay: 100ms;
  cursor: pointer;
  color: #dcfce7;
  background-color: #4d7c0f;
}

.c0:hover {
  background-color: #65a30d;
}

.c0:disabled {
  background-color: #eee;
  color: #aaa;
  cursor: not-allowed;
}

<body>
    <div>
      <button
        class="c0"
      >
        Hello world!
      </button>
    </div>
  </body>,
  "container": .c0 {
  border: none;
  outline: none;
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  font-size: 0.875rem;
  font-weight: 600;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-transition-delay: 100ms;
  transition-delay: 100ms;
  cursor: pointer;
  color: #dcfce7;
  background-color: #4d7c0f;
}

.c0:hover {
  background-color: #65a30d;
}

.c0:disabled {
  background-color: #eee;
  color: #aaa;
  cursor: not-allowed;
}

<div>
    <button
      class="c0"
    >
      Hello world!
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
